editor/components/block-list/with-hover-areas.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * WordPress dependencies
+ */
+import { Component, findDOMNode } from '@wordpress/element';
+
+/**
+ * Module constants
+ */
+const HOVER_AREA_SIZE = 120;
+
+const withHoverAreas = ( WrappedComponent ) => {
+	class WithHoverAreasComponent extends Component {
+		constructor() {
+			super( ...arguments );
+			this.state = {
+				hoverArea: null,
```

Comment:
I can easily add an `isHovered` state to this component to remove it from the `block.js` component but it seems that this state is slightly more complex than just keeping a boolean on whether the block is hovered or not. Might be good to do but I'll defer to @aduth here. I think you have more knowledge about the edge cases there.